code/modules/reagents/Chemistry-Machinery.dm

```diff
@@ -183,17 +184,15 @@ USE THIS CHEMISTRY DISPENSER FOR MAPS SO THEY START AT 100 ENERGY
 
 	if(href_list["amount"])
 		if(href_list["amount"] == "0")
-			var/num = input("Enter desired output amount", "Amount", "30") as num
+			var/num = input("Enter desired output amount", "Amount", useramount) as num
 			if (num)
 				amount = round(text2num(num), 5)
+				useramount = amount
 				custom = 1
 		else
 			custom = 0
 			amount = round(text2num(href_list["amount"]), 5) // round to nearest 5
-		if (amount < 5) // Since the user can actually type the commands himself, some sanity checking
-			amount = 5
-		if (amount > 100)
-			amount = 100
+		amount = Clamp(amount, 5, 100) // Since the user can actually type the commands himself, some sanity checking

```

Comment:
You should add the useramount setting after this, for sanity.
Try

    amount = Clamp(amount, 5, 100)
    if(custom)
        useramount = amount